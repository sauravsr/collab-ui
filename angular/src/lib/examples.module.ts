import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';

import { AlertExamplesModule } from './alert/examples/examples.module';
import { BadgeExamplesModule } from './badge/examples/examples.module';
import { ButtonExamplesModule } from './button/examples/examples.module';
import { CheckboxExamplesModule } from './checkbox/examples/examples.module';
import { RadioExamplesModule } from './radio/examples/examples.module';
import { InputExamplesModule } from './input/examples/examples.module';
import { SelectExamplesModule } from './select/examples/select-examples.module';
import { ListExamplesModule } from './list/examples/list-examples.module';
import { ListItemExamplesModule } from './list-item/examples/list-item-examples.module';

@NgModule({
  imports: [
    AlertExamplesModule,
    BadgeExamplesModule,
    ButtonExamplesModule,
    CheckboxExamplesModule,
    RadioExamplesModule,
    InputExamplesModule,
    SelectExamplesModule,
    ListExamplesModule,
    ListItemExamplesModule,
  ],
  exports: [
    AlertExamplesModule,
    BadgeExamplesModule,
    ButtonExamplesModule,
    CheckboxExamplesModule,
    RadioExamplesModule,
    InputExamplesModule,
    SelectExamplesModule,
    ListExamplesModule,
    ListItemExamplesModule
  ],
  declarations: [],
  providers: [],
  schemas: [ CUSTOM_ELEMENTS_SCHEMA ],

})
export class ExamplesModule { }
