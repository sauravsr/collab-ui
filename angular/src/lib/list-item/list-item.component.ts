import { Component, OnInit, Input, HostBinding, HostListener, ElementRef,
  Output, EventEmitter, AfterViewInit, ChangeDetectorRef, AfterContentChecked } from '@angular/core';
import { uniqueId } from 'lodash';

export class OptionSelectionChange {
  constructor(public source: ListItemComponent) {}
}

@Component({
  selector: 'a[cui-list-item], div[cui-list-item], div[cui-select-option]',
  exportAs: 'cuiListItem',
  template: `
      <ng-container *ngIf="isSelectOption">
        <ng-container *ngIf="!isMulti">
          <cui-list-item-section key="child-0" position='center'>
              <ng-container *ngIf="label; else content">{{label}}</ng-container>
              <ng-template #content>
                  <ng-content></ng-content>
              </ng-template>
          </cui-list-item-section>
          <cui-list-item-section key="child-1" position='right'>
              <cui-icon *ngIf="active" name="check_20" color="blue"></cui-icon>
          </cui-list-item-section>
        </ng-container>

        <ng-container *ngIf="isMulti">
          <cui-checkbox
            name="{{id}}-checkbox"
            value="{{label}}"
            label="{{label}}"
            [(checkStatus)]="checkStatus"
            htmlId="{{id}}-checkbox">
          </cui-checkbox>
        </ng-container>
      </ng-container>

      <ng-container *ngIf="!isSelectOption">
        <div *ngIf="label; else content">{{label}}</div>

        <ng-template #content>
          <ng-content></ng-content>
        </ng-template>
      </ng-container>
  `
})
export class ListItemComponent implements OnInit, AfterViewInit, AfterContentChecked {
  checkedValues: string[] = [''];
  checkStatus: boolean;

  readonly isSelectOption: boolean = this._hasHostAttributes('cui-select-option');

  constructor(
    private el: ElementRef,
    private _changeDetectorRef: ChangeDetectorRef
    ) { }

  /** @option Active prop to help determine styles | false */
  @Input() active = false;
  /** @option class Optional css class name | '' */
  @Input() class = '';
  /** @option Disabled attribute for ListItem to determine styles | false */
  @Input() disabled = false;
  /** @option Sets ListItem id | null */
  @HostBinding('attr.id') @Input() id: string = uniqueId('cui-list-item-');
  /** @option Determines if is part of a multi select | false */
  @Input() isMulti = false;
  /** @option Determines if ListItem is clickable | false */
  @Input() isReadOnly = false;
  /** @option ListItem label text | '' */
  @Input() label = null;
  /** @option external link associated input | '' */
  @Input() link = '';
  /** @option ListItem ref name | 'navLink' */
  @HostBinding('attr.role') @Input() role = 'listItem';
  /** @option Prop that controls whether to show separator or not | false */
  @Input() separator = false;
  /** @option ListItem Title | '' */
  @Input() title = '';
  /** @option ListItem size | '' */
  @Input() type = '';

  @Output() selectionChange = new EventEmitter<OptionSelectionChange>();

  @HostBinding('attr.href') get myHref(): string {
    return (this.link && this._getHostElement().localName === 'a') ? this.link : null;
  }

  @HostBinding('class') get className(): string {
    return 'cui-list-item' +
    `${(this.type && ` cui-list-item--${this.type}`) || ''}` +
    `${(this.active && ` active`) || ''}` +
    `${(this.disabled && ` disabled`) || ''}` +
    `${(this.isReadOnly && ` cui-list-item--read-only`) || ''}` +
    `${(this.separator && ` cui-list-item--separator`) || ''}` +
    `${(this.class && ` ${this.class}`) || ''}`;
  }

  @HostBinding('attr.title') get theTitle() { return this.title || this.label; }

  @HostListener('click', ['$event.target']) handleClick = event => {
    if (this.isReadOnly) {
      event.stopImmediatePropagation();
    } else {
      this._selectViaInteraction();
    }
   }

  ngOnInit() {
    if (this.type && !this._isTypeOptionValid()) {
      throw new Error(`cui-list-item: ListItem type option must be one of the following:
        small, large, xlarge, space, header, 36, 52, 60`);
    }
  }

  ngAfterViewInit () {
    if (!this.label) {
      throw new Error('cui-list-item: Label property is required');
    }
  }

  ngAfterContentChecked () {}

  select(): void {
    if (!this.active) {
      this.active = true;
      this._changeDetectorRef.markForCheck();
      this._emitSelectionChangeEvent();
    }
  }

  /** Deselects the option. */
  deselect(): void {
    if (this.active) {
      this.active = false;
      this._changeDetectorRef.markForCheck();
      this._emitSelectionChangeEvent();
    }
  }

  _selectViaInteraction(): void {
    if (!this.disabled) {
      this.active = !this.active;
      this.checkStatus = this.active;
      this._changeDetectorRef.markForCheck();
      this._emitSelectionChangeEvent();
    }
  }

  _emitSelectionChangeEvent () {
    return this.selectionChange.emit(new OptionSelectionChange(this));
  }

  _isTypeOptionValid = () => (
    ['', 'small', 'large', 'xlarge', 'space', 'header', 36, 52, 60].includes(this.type)
  )

  _getHostElement() {
    return this.el.nativeElement;
  }

  /** Gets whether the button has one of the given attributes. */
  _hasHostAttributes(...attributes: string[]) {
    return attributes.some(attribute => this._getHostElement().hasAttribute(attribute));
  }
}
