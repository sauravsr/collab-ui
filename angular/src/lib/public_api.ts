/*
 * Public API Surface of collab-ui-angular
 */

export * from './alert';
export * from './badge';
export * from './button';
export * from './checkbox';
export * from './icon';
export * from './input';
export * from './input-error';
export * from './input-helper';
export * from './label';
export * from './list';
export * from './list-item';
export * from './list-item-section';
export * from './loading';
export * from './modal';
export * from './modal-body';
export * from './modal-header';
export * from './modal-footer';
export * from './radio';
export * from './select';
export * from './spinner';
export * from './tabs';
