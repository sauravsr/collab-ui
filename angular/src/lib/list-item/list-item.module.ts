import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ListItemComponent } from './list-item.component';
import { IconModule } from '../icon';
import { ListItemSectionModule} from '../list-item-section';
import { CheckboxModule } from '../checkbox';

@NgModule({
  imports: [
    CheckboxModule,
    CommonModule,
    IconModule,
    ListItemSectionModule
  ],
  declarations: [ListItemComponent],
  exports: [ListItemComponent],
  schemas: [CUSTOM_ELEMENTS_SCHEMA]
})
export class ListItemModule { }
