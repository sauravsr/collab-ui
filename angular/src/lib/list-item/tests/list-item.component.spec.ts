import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ListItemComponent } from '../list-item.component';
import { CheckboxModule, IconModule, ListItemSectionModule } from 'src/lib/public_api';

describe('ListItemComponent', () => {
  let component: ListItemComponent;
  let fixture: ComponentFixture<ListItemComponent>;
  let listItemNativeElement: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ ListItemComponent ],
      imports: [CheckboxModule, IconModule, ListItemSectionModule]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ListItemComponent);
    component = fixture.componentInstance;
    component.label = 'myLabel';
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render with default class name', () => {
    component.class = 'custom-class';
    fixture.detectChanges();

    listItemNativeElement = fixture.nativeElement;
    expect(listItemNativeElement.className).toContain(component.class);
  });

  it('should render with disabled class when defined', () => {
    component.disabled = true;
    fixture.detectChanges();

    listItemNativeElement = fixture.nativeElement;
    expect(listItemNativeElement.className).toContain('disabled');
  });

  it('should render as readOnly when defined', () => {
    component.isReadOnly = true;
    fixture.detectChanges();

    listItemNativeElement = fixture.nativeElement;
    expect(listItemNativeElement.className).toContain('cui-list-item--read-only');
  });

  it('should render the defined type', () => {
    component.type = 'small';
    fixture.detectChanges();

    listItemNativeElement = fixture.nativeElement;
    expect(listItemNativeElement.className).toContain( `cui-list-item--${component.type}`);
  });
});
