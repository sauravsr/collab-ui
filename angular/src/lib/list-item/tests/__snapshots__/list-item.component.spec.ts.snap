// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemComponent should create and match snapshot 1`] = `
<a
  _changeDetectorRef={[Function ViewRef_]}
  _isTypeOptionValid={[Function Function]}
  active="false"
  checkedValues={[Function Array]}
  class=""
  disabled="false"
  el={[Function ElementRef]}
  handleClick={[Function Function]}
  id={[Function String]}
  isMulti="false"
  isReadOnly="false"
  isSelectOption="false"
  label={[Function String]}
  link=""
  role={[Function String]}
  selectionChange={[Function EventEmitter]}
  separator="false"
  title=""
  type=""
>
  <div>
    myLabel
  </div>
</a>
`;
